### ISSUE TITLE
Loading 16bit png files fails with pillow 8.3

### ISSUE THREAD

**Issue by martinResearch** [imageio/imageio#655]
Loading 16bit png files with pillow 8.3 broken in imagio 2.9.0

To reproduce: 

python 3.8.6
numpy==1.19.3
imageio==2.9.0
pillow==8.3

This will fail in pillow 8.3 while it work with pillow 8.2

```
import imageio
import numpy as np

im1 = np.zeros((3, 3), np.uint16)
im1[0, 0] = 1000
imageio.imsave("test.png", im1)
im2 = imageio.imread("test.png", "PNG-PIL")
```
error message 
```
 TypeError: __array__() takes 1 positional argument but 2 were given
   File  ...\lib\site-packages\imageio\plug ins\pillow.py", line 790, in pil_get_frame
   frame = np.array(frame, dtype=dtype)
```
while it  works using freeimage `im2 = imageio.imread('test.png', 'PNG-FI')`or using pillow 8.2

changing pillow.py from 
`frame = np.array(frame, dtype=dtype)`
to 
`frame = np.array(frame).astype(dtype=dtype)`
seems to fix the problem.
 It looks like the same issue as https://github.com/python-pillow/Pillow/issues/5571
so imageio will probably need to update pillow's dependency once fixed.

**Comment by FirefoxMetzger** [comment]
Yep, this is a bug in the pillow backend (as you have already pointed out).

My suggestion is to simply freeze the version to pillow < 8.3 for imageIO v2.9 instead of implementing a patch. I'm personally not a fan of patching upstream bugs downstream. Especially considering that pillow is quite actively developed, so a fix should come around quite quickly. We are also working towards v2.10 and by the time we release it, pillow will have fixed this bug so our v2.10 can likely depend on pillow v8.3 aka. v8.3.1

That said, it also depends on @almarklein s thoughts on this.

**Comment by almarklein** [comment]
I agree. We should revisit this once we're nearing v2.10.
